google/cloud/bigtable/internal/bulk_mutator.cc

```diff
@@ -147,11 +152,19 @@ void BulkMutator::FinishRequest() {
 
 std::vector<FailedMutation> BulkMutator::ExtractFinalFailures() {
   std::vector<FailedMutation> result(std::move(failures_));
-  google::rpc::Status ok_status;
-  ok_status.set_code(grpc::StatusCode::OK);
+  // These are most likely an effect of a broken stream. We do not know
+  // their error code and there are not going to be any more retries. Report
+  // them as UNKNOWN in the failure list.
+  google::rpc::Status status;
+  status.set_code(grpc::StatusCode::UNKNOWN);
+  status.set_message(
+      "Never got a confirmation for this mutation. Most likely stream was "
+      "broken before its status was sent.");
+  int idx = 0;
   for (auto& mutation : *pending_mutations_.mutable_entries()) {
-    result.emplace_back(
-        FailedMutation(SingleRowMutation(std::move(mutation)), ok_status));
+    auto &annotation = pending_annotations_[idx++];
```

Comment:
Does this need to be `auto &`? I'm worried that if `pending_annotations_` reallocates that you'll have undefined behavior. Of course, it's possible that this does not happen, but better safe than sorry.

Reply:
Not by default, no. I was asking in this specific case since you are more familiar with the flow of the code than I am.

Reply:
How about now? I removed the reference.

Reply:
That's a good solution. Keep in mind that I am a new grad so I am asking to learn, not to make frivolous reviews.

Reply:
Hey, I think it's good to point out everything that seems wrong or suboptimal.

I've learned tons of stuff by people showing me in reviews how my code could be improved, so I appreciate that you challenge what I wrote.